analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Constraints/NumberConstraint.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Numerics;
+
+namespace SonarAnalyzer.SymbolicExecution.Constraints;
+
+public sealed class NumberConstraint : SymbolicConstraint
+{
+    public static readonly NumberConstraint Empty = new(null, null);
+    public static readonly NumberConstraint Zero = new(0, 0);
+    public static readonly NumberConstraint One = new(1, 1);
+
+    public BigInteger? Min { get; }
+    public BigInteger? Max { get; }
+
+    public override SymbolicConstraint Opposite =>
+        null;
+
+    private NumberConstraint(BigInteger? min, BigInteger? max) : base(ConstraintKind.Number)
+    {
+        if (min.HasValue && max.HasValue && min.Value > max.Value)
+        {
+            Max = min;  // Swap
+            Min = max;
+        }
+        else
+        {
+            Min = min;
+            Max = max;
+        }
+    }
+
+    public static NumberConstraint From(BigInteger value)
+    {
+        if (value.IsZero)
+        {
+            return Zero;
+        }
+        else if (value.IsOne)
+        {
+            return One;
+        }
+        else
+        {
+            return new(value, value);
+        }
+    }
+
+    public static NumberConstraint From(BigInteger? min, BigInteger? max)
```

Comment:
Let's rename the parameters (also change them for the private constructor). I think `boundary1` and ` boundary2` are appropriate.

Reply:
boundary1 and boundary2 doesn't work for infinite ranges, like `> 0` where min=0, max=null